core/test/unit/api/v2/session_spec.js

```diff
@@ -0,0 +1,120 @@
+const should = require('should');
+const sinon = require('sinon');
+
+const models = require('../../../../server/models');
+const {UnauthorizedError} = require('../../../../server/lib/common/errors');
+
+const sessionController = require('../../../../server/api/v2/session');
+const sessionServiceMiddleware = require('../../../../server/services/auth/session/middleware');
+
+describe('Session controller', function () {
+    let sandbox;
+
+    before(function () {
+        models.init();
+        sandbox = sinon.sandbox.create();
+    });
+
+    afterEach(function () {
+        sandbox.restore();
+    });
+
+    it('exports an add method', function () {
+        should.equal(typeof sessionController.add, 'function');
+    });
+    it('exports an delete method', function () {
+        should.equal(typeof sessionController.delete, 'function');
+    });
+
+    describe('#add', function () {
+        it('throws an UnauthorizedError if the object and options is missing', function () {
```

Comment:
hmmm - we can always move it during a cleanup time?

Reply:
yeah i know, just want to avoid extra work. 
if the shared frame is ready to merge, we should probably immediataly convert the session controller as first thing?

Reply:
either way we have to do the work, just now or later - i'd prefer to leave this for now as it'll have a knockon affect on other stuff - also it's a very weird controller, so it's might be a good idea to wait for the framework to settle a bit?

Reply:
okay leave as it and merge?
i just need to double check the whole PR quickly

Reply:
okay - when you are happy let me know - i will reword the test and merge